sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -840,10 +869,11 @@ public CosmosAsyncClient buildAsyncClient() {
      * @return CosmosClient
      */
     public CosmosClient buildClient() {
-
         validateConfig();
         buildConnectionPolicy();
-        return new CosmosClient(this);
+        CosmosClient cosmosClient = new CosmosClient(this);
+        cosmosClient.openConnectionsAndInitCaches();
+        return cosmosClient;
```

Comment:
The total start up time and config used should be logged. A PR to do so (not on this branch/PR) here: https://github.com/Azure/azure-sdk-for-java/pull/33153 (some questions on the that PR for use of core SDK's LoggerClient vs direct log4j logger factory).

Reply:
Done